src/compiler/types.ts

```diff
@@ -623,7 +623,7 @@ namespace ts {
         /*@internal*/ autoGenerateKind?: GeneratedIdentifierKind; // Specifies whether to auto-generate the text for an identifier.
         /*@internal*/ autoGenerateId?: number;                    // Ensures unique generated identifiers get unique names, but clones get the same name.
         isInJSDocNamespace?: boolean;                             // if the node is a member in a JSDoc namespace
-        /*@internal*/ typeArguments?: NodeArray<TypeNode>;        // Only defined on synthesized nodes. Though not syntactically valid, used in emitting diagnostics.
+        /*@internal*/ typeArguments?: NodeArray<TypeNode | TypeParameterDeclaration>; // Only defined on synthesized nodes. Though not syntactically valid, used in emitting diagnostics.

```

Comment:
Should this comment now say it is used for emitting diagnostics and quickinfo? Are there any other cases where this might be used? Do we have some way of asserting this isn't misused in the emitter (eg: assert it's undefined when emitting a declaration file)?

Reply:
Re: the comment. Sure, done. Re: asserting misuse: It is only set when the caller to the node builder sets an explicit "I want type parameters on Identifiers" flag. It's not an easy thing to get by accident.